sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -121,6 +131,37 @@ protected void beforeTest() {
     protected void afterTest() {
     }
 
+    /**
+     * Returns a list of {@link HttpClient HttpClients} that should be tested.
+     * @return A list of {@link HttpClient HttpClients} to be tested.
+     */
+    protected static Stream<HttpClient> getHttpClients() {
+        if (testMode == TestMode.PLAYBACK) {
+            // Call to @MethodSource method happens @BeforeEach call, so the interceptorManager is
+            // not yet initialized. So, playbackClient will not be available until later.
+            return Stream.of(new HttpClient[]{null});
+        }
+        return HttpClientProviders.getAllHttpClients().stream().filter(TestBase::shouldClientBeTested);
+    }
+
+    /**
+     * Filter out the http clients needs to run in test framework
+     * @param client Http client needs to check
+     * @return Boolean indicates whether filters out the client or not.
+     */
+    protected static boolean shouldClientBeTested(HttpClient client) {
+        // This is for when we decide to filter some http clients based on some criteria
+        // to reduce the time take to run the tests.
```

Comment:
Replace this comment with what this method is doing. We are using Netty http client as default if env variable is not set. If it's set to ALL, all HttpClients in the classpath will be tested else the name of the HttpClient class should match env variable.

Reply:
Update the comment here.